org.lflang/src/org/lflang/validation/LFValidator.xtend

```diff
@@ -749,6 +749,19 @@ class LFValidator extends BaseLFValidator {
             if (param.inferredType.isUndefined()) {
                 error("Type declaration missing.", Literals.PARAMETER__TYPE)
             }
+        } 
+        if(this.target == Target.CPP) {
+            val container = param.eContainer
+            val reactor = container as Reactor
+            if(reactor.isMain){ 
+                // we need to check for the cli parameters that are always taken
+                val myList = #['t', 'threads', 'o', 'timeout', 'k', 'keepalive', 
```

Comment:
I think it would be better to define these in the `Targets` class and to generalize this check (not only for C++).

Reply:
Just for reference, the Rust target avoids these naming conflicts by prefixing user-defined parameter names with `main-`. So if a user defines a `threads` parameter it ends up on the CLI as `--main-threads`. Maybe we should align both targets on either behavior.

Reply:
> Just for reference, the Rust target avoids these naming conflicts by prefixing user-defined parameter names with `main-`. So if a user defines a `threads` parameter it ends up on the CLI as `--main-threads`. Maybe we should align both targets on either behavior.

That sound like a good idea. 
I was not sure what our guideline is on that subject. 
In combination with @lhstrh  comment I will clean this up.